src/sinks/gcp/pubsub.rs

```diff
@@ -166,24 +155,40 @@ impl HttpSink for PubsubSink {
         Some(json!({ "data": base64::encode(&json) }))
     }
 
-    fn build_request(&self, events: Self::Output) -> http::Request<Vec<u8>> {
+    fn build_request(&self, events: Self::Output) -> Request<Vec<u8>> {
         let body = json!({ "messages": events });
         let body = serde_json::to_vec(&body).unwrap();
 
-        let mut builder = hyper::Request::builder();
-        builder.method(Method::POST);
-        builder.uri(self.uri(":publish").unwrap());
-        builder.header("Content-Type", "application/json");
+        let builder = Request::builder()
+            .method(Method::POST)
+            .uri(self.uri(":publish").unwrap())
+            .header("Content-Type", "application/json");
 
         let mut request = builder.body(body).unwrap();
         if let Some(creds) = &self.creds {
-            creds.apply(&mut request);
+            creds.apply2(&mut request);
         }
 
         request
     }
 }
 
+async fn healthcheck(
+    cx: SinkContext,
+    uri: Uri,
+    tls: TlsSettings,
+    creds: Option<GcpCredentials>,
+) -> crate::Result<()> {
+    let mut request = Request::get(uri).body(Body::empty()).unwrap();
+    if let Some(creds) = creds.as_ref() {
+        creds.apply2(&mut request);
+    }
+
+    let mut client = HttpClient::new(cx.resolver(), tls.clone())?;
+    let response = client.send(request).await?;
+    healthcheck_response2(creds, HealthcheckError::TopicNotFound.into())(response)
+}
```

Comment:
async function require static lifetime... 
`healthcheck(&self, ...` require lifetime, so I decide move it out of struct.

I can return it back, but then this will be like this:
```rust
fn healthcheck(
    &self,
    cx: SinkContext,
) -> crate::Result<Pin<Box<Future<Output = crate::Result<()>>>>>
    let uri = self.uri("")?
    let tls_settings = TlsSettings::from_options(&self.tls)?
    let creds = self.creds.clone();

    Ok(Box::pin(async move {
        let mut request = Request::get(uri).body(Body::empty()).unwrap();
        if let Some(creds) = creds.as_ref() {
            creds.apply2(&mut request);
        }

        let mut client = HttpClient::new(cx.resolver(), tls.clone())?;
        let response = client.send(request).await?;
        healthcheck_response2(creds, HealthcheckError::TopicNotFound.into())(response)
    }))
}
```

Should we return it back?

Reply:
I think you're right. While it is good to be able to reuse the struct to avoid having to pass additional parameters, I do prefer the more obvious code that results from the `async fn`.